### PR TITLE
Rewrite

### PR DESCRIPTION
**Pull request by dutsmiller**
Significant changes
- Update to helm chart 3.x
- Optionally manage identities within module (can still be created separately).

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -10,12 +10,39 @@ variable "aks_principal_id" {
...
 }
 
 variable "helm_chart_version" {
   description = "Azure AD pod identity helm chart version"
   type        = string
-  default     = "2.0.0"
+  default     = "3.0.1"
 }
+
+variable "intstall_crds" {
```
Comment (b-odonoghue) [inline variables.tf]: @dutsmiller this should be `install_crds` vs `int`stall_crds

Comment (b-odonoghue) [reply variables.tf:23]: I have added a commit to fix this

File: providers.tf [hunk 2]
```diff
@@ -2,6 +2,6 @@ terraform {
   required_version = ">= 0.13.0"
 
   required_providers {
-    helm       = ">= 1.2.4"
+    helm = ">= 2.0.0"
```
Comment (fabiendelpierre) [inline providers.tf]: Please update this to TF 0.14 syntax (with the added `source` argument).

File: variables.tf [hunk 3]
```diff
@@ -3,19 +3,46 @@ variable "aks_node_resource_group" {
...
+    namespace   = string
+    name        = string
+    client_id   = string
+    resource_id = string
+  }))
+  default = null
+}
+
+variable "enable_kubenet_plugin" {
+  description = "Enable feature when AKS cluster is uses Kubenet for CNI, leave default if use AzureCNI"
```
Comment (fabiendelpierre) [inline variables.tf]: Grammar fix? 😄

File: variables.tf [hunk 4]
```diff
@@ -3,19 +3,46 @@ variable "aks_node_resource_group" {
...
+}
+
+variable "install_crds" {
+  description = "Install CRDs"
+  type        = bool
+  default     = true
+}
+
+variable "identities" {
+  description = "Azure identites to be configured"
```
Comment (fabiendelpierre) [inline variables.tf]: Fix spelling of "identities"

**Review by fabiendelpierre — CHANGES_REQUESTED** [review]

**Review by fabiendelpierre — APPROVED** [review]

**Pull request merged**